common/lib/xmodule/xmodule/modulestore/xml.py

```diff
@@ -504,23 +526,40 @@ def get_policy(usage_id):
             if isinstance(course_descriptor, ErrorDescriptor):
                 return course_descriptor
 
-            # NOTE: The descriptors end up loading somewhat bottom up, which
-            # breaks metadata inheritance via get_children().  Instead
-            # (actually, in addition to, for now), we do a final inheritance pass
-            # after we have the course descriptor.
-            compute_inherited_metadata(course_descriptor)
-
-            # now import all pieces of course_info which is expected to be stored
-            # in <content_dir>/info or <content_dir>/info/<url_name>
-            self.load_extra_content(system, course_descriptor, 'course_info', self.data_dir / course_dir / 'info', course_dir, url_name)
+            if not library:
+                # NOTE: The descriptors end up loading somewhat bottom up, which
+                # breaks metadata inheritance via get_children().  Instead
+                # (actually, in addition to, for now), we do a final inheritance pass
+                # after we have the course descriptor.
+                compute_inherited_metadata(course_descriptor)
```

Comment:
@cpennington I spent a good chunk of time trying to track the root cause of this down, but I'm having difficulty. I know /why/ the metadata inheritance fails. But I don't know the why of that why :)

Library.xblock_kvs.inherited_settings does not exist because xblock_kvs in that instance is a DictKeyValueStore, rather than an InheritanceKeyValueStore. As much as I can tell, XModules use InheritanceKeyValueStore, whereas XBlocks use DictKeyValueStore, and since Library is an XBlock, it fails here.

I tried replacing DictKeyValueStore in the importer with InheritanceKeyValueStore, but this appeared to create subtle problems that failed tests, even though it allowed the import to continue. So I imagine there's a reason why DictKeyValueStore is used in xml_importer instead. Could you explain what the Metadata actually is, and what I should look into for solving this problem?